applications/nrf_desktop/src/modules/Kconfig.profiler_sync

```diff
@@ -0,0 +1,41 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+menu "Profiler sync"
+
+config DESKTOP_PROFILER_SYNC_GPIO_ENABLE
+	bool "Enable profiler sync based on GPIO"
+	depends on GPIO
+	depends on DESKTOP_EVENT_MANAGER_PROFILER_ENABLED
+	help
+	  This option can be used to synchronize timestamps for profiler events
+	  on two devices.
+
+	  When this option is enabled nRF Desktop devices generate profiler
+	  event (sync_event) that is used for synchronization. The central
+	  generates square wave using GPIO, the peripheral reacts on the edges.
+	  Make sure that proper pins of the central and the peripheral are
+	  connected together.
+
+if DESKTOP_PROFILER_SYNC_GPIO_ENABLE
+
+config DESKTOP_PROFILER_SYNC_GPIO_PORT
+	int "Sync GPIO port"
+	range -1 1
+	default -1
```

Comment:
You are putting a range `-1 1` here, but code in profiler_sync.c:82-83 has:
```
BUILD_ASSERT(CONFIG_DESKTOP_PROFILER_SYNC_GPIO_PORT != -1,
	     "Please configure sync GPIO pin and port");
```
which will cause a compile error if the default value of `-1` is used.

Secondly, what does a port value of `-1` actually indicate ?
There is no help text.

Reply:
This is actually done to make sure that user will not use default value, because the value must be set properly for every test setup.

To synchronize timestamps we need to generate synchronization events on both devices at the same time. The devices synchronize using 1 GPIO pin each. These pins must be connected together (details described in the DESKTOP_PROFILER_SYNC_GPIO_ENABLE help).

Reply:
By why would a default value of `0` not be ok ?
From current text, it is not clear to me if one device must use `0` and the other device `1`, or if they must use same value.

I understand that the configured pins must be correctly connected between the devices, but I don't see that this fact prevents the use of a default pin.
Secondly, pin assignment should be done in DTS, afaik.
@mbolivar-nordic could you join, pin assignments should go to DTS, right ?

Reply:
Short explanation of module's purpose:
We got 2 wirelessly connected devices (devA and devB). On both devices we log profiler events (we log such event when something in the application happens). Then we measure time intervals between the profiler events to check performance of the application.

In case all the events are from one device (for example devA), the timestamps are generated using the same clock source - no additional synchronization is needed.

When we want to calculate times between events on two various devices we need to take into account that we use 2 various clock sources for generating the timestamps and we need to synchronize these timestamps. For that purpose we need to log profiler event exactly at the same time on both device - this is done by the profiler_sync module.

To achieve this, the devices are be connected using physical wire. DevA changes state of the pin connected to the wire and logs profiler event. DevB logs profiler event when it notices state change of the pin that is connected to another end of the wire. The event is logged on both devices at the same time.

Reply:
I for putting 0 as the default. It's actually the same as pin - you have to specify the correct value there too but there is no fancy guard protecting against idiots.

Reply:
> Short explanation of module's purpose:

Nice explanation, and this was also my understanding already.

But it doesn't answer my question.

Why is it not possible for both devices to have a default value of `0` or `1` ?
After all, we should try to help users by providing a default set of values that works out-of-the box by simply following a guide.

Like:
> On nrf desktop, connect pin m, device A to pin n, device B, and then enable profiler sync.
>
> If you wish to use other pins, please adjust settings accordingly

Reply:
The selected pin and port depends on board (for example on one board the pin/port combination could be used for some other purpose). I wanted to make sure that user is aware that he must define proper pin/port for used boards (that user is aware which pin/port is used by the module).

After discussion I decided that we could simply set this to some fixed, but proper value (like P0.0), inform about this in documentation and add the information that user can change it to other value (to match his setup). Will this be ok for you ?

Reply:
> The selected pin and port depends on board (for example on one board the pin/port combination could be used for some other purpose). 

For default setups, that could be something to place in overlay files for that particular board, so that it does have a reasonable default pin.
Of course there is always a risk of conflict if user enables this on the board, after he has already assigned those default pins, but hat is a risk for any default values in NCS.

Regarding pin assignment, I would still like to get @mbolivar-nordic opinion regarding this being in DT.

> After discussion I decided that we could simply set this to some fixed, but proper value (like P0.0), inform about this in documentation and add the information that user can change it to other value (to match his setup). Will this be ok ?

That sounds reasonable.

Reply:
Hi @tejlmand , @mbolivar-nordic , I think it is a bit pointless to put this in the DT. First of all to put anything in the DT you need to have a correct binding. We would need to create a binding for the profiler sync pin or something. And this seems like an overshot as it won't be used for anything aside nrf_desktop and even we will use it only to get the correct times between boards.
I know that general rule is to put HW definition in the DT but seriously it is not to go through the effort in this case.

Reply:
I agree with @tejlmand that this should go into DT.

> We would need to create a binding for the profiler sync pin or something.

If this were true, I would understand your concern. But it is not correct as far as I can tell. The existing gpio-keys binding seems to serve this purpose, based on @MarekPieta 's description of the module's purpose. Did I misunderstand something here?

Reply:
This GPIO is not a key? Obviously we can use any random binding where we can stick the GPIO port and pin but would it really be better then simply put the value in the kconfig? Having it as Marek put it would be much simpler to use. Please don't over-complicate it ^_^"

Reply:
The module is used only by nrf_desktop for profiling.

Using Kconfig here is way more convenient from our perspective (we do not need to modify/overlay DTS for every board separately). It also makes us able to easily change pin used for a given test (we already use the menuconfig to enable profiler).

Reply:
OK, if it is for internal use only and not expected to be user-visible, I won't block.